src/encoded/static/components/top_hits/multi/search.js

```diff
@@ -13,7 +13,11 @@ import QUERIES from './constants';
 * dropdown using the associated custom results component.
 */
 const Search = ({ children }) => {
+    // User input.
     const [input, setInput] = useState('');
+    // Reference to latest input for use in callback.
+    const inputRef = useRef();
+    inputRef.current = input;
```

Comment:
I didn’t think about this before, but I think these two lines can be combined into:
```
const inputRef = useRef(input);
```
I’m about 80% sure that’s equivalent to what you have.

Reply:
Hmm, this doesn't seem to work (tests break). I wonder if I should wrap the assignment in useEffect?
```javascript
useEffect(() => {
        inputRef.current = input;
    }, [input]);
```